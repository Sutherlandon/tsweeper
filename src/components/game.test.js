import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Game from './game';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Game />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('calculates cell values properly', () => {
  const game = renderer.create(<Game />);
  let tree = game.toJSON();
  expect(tree).toMatchSnapshot();
});
