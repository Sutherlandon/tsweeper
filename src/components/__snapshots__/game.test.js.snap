// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculates cell values properly 1`] = `
<div
  className="terminal"
>
  <div
    className="line"
  >
    Welcome to T
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
    Sweeper a text based minesweeper.
  </div>
  <br />
  <div
    className="line index"
  >
      0 1 2 3 4 5 6 7 8
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      0
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      1
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      2
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      3
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      4
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      5
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      6
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      7
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="index"
    >
      8
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      - 
    </span>
  </div>
  <br />
  <div
    className="line"
  >
    Commands: (1) Reveal, (2) Flag
  </div>
  <div
    className="line"
    style={
      Object {
        "color": "red",
      }
    }
  />
  <div
    className="line"
  >
    <form
      onSubmit={[Function]}
    >
      &gt; 
      <input
        className="terminal-input"
        onChange={[Function]}
        type="number"
        value=""
      />
    </form>
  </div>
</div>
`;
